code/modules/vehicles/mecha/mech_bay.dm

```diff
@@ -52,27 +61,86 @@
 	if(in_range(user, src) || isobserver(user))
 		. += span_notice("The status display reads: Recharge power <b>[siunit(recharge_power, "W", 1)]</b>.")
 
-/obj/machinery/mech_bay_recharge_port/process(seconds_per_tick)
+/obj/machinery/mech_bay_recharge_port/power_change()
+	if(!(machine_stat & NOPOWER))
+		begin_processing()
+	return ..()
```

Comment:
Surely you want to check *after* parent call, right? 

Parent handles setting the nopower flag on or off